pybikes/bysykkelgraphql.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+import json
+
+import requests
+
+from . import utils
+from .base import BikeShareStation, BikeShareSystem
+
+
+class BySykkelGraphql(BikeShareSystem):
+
+    authed = False
+
+    meta = {
+        'system': 'BySykkelGraphql',
+        'company': ['Urban Infrastructure Partner'],
+    }
+
+    def __init__(self, tag, meta, feed_url):
+        super(BySykkelGraphql, self).__init__(tag, meta)
+        self.feed_url = feed_url
+
+    def update(self, scraper=None):
+        if scraper is None:
+            scraper = utils.PyBikesScraper()
+        scraper.headers['Content-Type'] = 'application/json'
+
+        self.stations = []
+        query = """
+query dockGroups($systemId: ID){
+  dockGroups(systemId: $systemId) {
+    id
+    title
+    state
+    address
+    coord {
+      lat
+      lng
+    }
+    docks {
+      id
+      state
+      vehicleDocked
+      availabilityState
+    }
+  }
+}
+"""
+
+        variables = {'systemId': self.meta['systemId']}
+
+        gql_data = {
+            'query': query,
+            'operationName': 'dockGroups',
+            'variables': variables
+        }
+
+        data = json.loads(
+            scraper.request(
+                self.feed_url,
+                data=json.dumps(gql_data).encode('utf-8'),
+                method='POST'))
```

Comment:
Does this request randomly fail to you too? I have had 2/5 of success running it. Others return a 5xx from their service

Reply:
Hmm. Works for me every time. Since you get a `5XX` response, I guess they get a message about the error; hopefully resulting in them fixing the issue.